File: molecule/default/tests/test_default.rb
# frozen_string_literal: true

describe mount('/efs') do
  it { should be_mounted }
  its('device') { should eq 'none' }
  its('type') { should eq 'aufs' }
end

describe file('/efs/test') do
  it { should exist }
end

describe command('ls /efs/*/info') do
  its('exit_status') { should eq 0 }
end

describe command('ls /efs/*/info/distro_name') do
  its('exit_status') { should eq 0 }
end

describe command('ls /efs/*/scons-cache') do
  its('exit_status') { should eq 0 }
end

File: molecule/default/tests/test_default.py
import os

import testinfra.utils.ansible_runner

testinfra_hosts = testinfra.utils.ansible_runner.AnsibleRunner(
    os.environ['MOLECULE_INVENTORY_FILE']
).get_hosts('all')


def test_mount(host):
    assert host.mount_point("/efs").exists
    assert host.file("/efs/test").exists

    cmd = host.run("ls /efs/*/info")
    assert cmd.succeeded
    cmd = host.run("ls /efs/*/info/distro_name")
    assert cmd.succeeded
    cmd = host.run("ls /efs/*/scons-cache")
    assert cmd.succeeded
